packages/modeling/src/measurements/measureBoundingBox.js

```diff
@@ -7,38 +7,61 @@ import * as geom2 from '../geometries/geom2/index.js'
 import * as geom3 from '../geometries/geom3/index.js'
 import * as path2 from '../geometries/path2/index.js'
 import * as poly3 from '../geometries/poly3/index.js'
+import * as slice from '../geometries/slice/index.js'
 
 const cache = new WeakMap()
 
 /*
- * Measure the min and max bounds of the given (path2) geometry.
- * @return {Array[]} the min and max bounds for the geometry
+ * Expand bounding box with a 2D point.
  */
-const measureBoundingBoxOfPath2 = (geometry) => {
-  let boundingBox = cache.get(geometry)
-  if (boundingBox) return boundingBox
-
-  const points = path2.toPoints(geometry)
-
-  let minPoint
-  if (points.length === 0) {
-    minPoint = vec2.create()
+const expand2 = (bbox, point) => {
+  if (bbox.length === 0) {
+    bbox[0] = vec3.fromVec2(vec3.create(), point)
+    bbox[1] = vec3.fromVec2(vec3.create(), point)
   } else {
-    minPoint = vec2.clone(points[0])
+    vec2.min(bbox[0], bbox[0], point)
+    vec2.max(bbox[1], bbox[1], point)
   }
-  let maxPoint = vec2.clone(minPoint)
-
-  points.forEach((point) => {
-    vec2.min(minPoint, minPoint, point)
-    vec2.max(maxPoint, maxPoint, point)
-  })
-  minPoint = [minPoint[0], minPoint[1], 0]
-  maxPoint = [maxPoint[0], maxPoint[1], 0]
+}
 
-  boundingBox = [minPoint, maxPoint]
+/*
+ * Expand bounding box with a 3D vertex.
+ */
+const expand3 = (bbox, vertex) => {
+  if (bbox.length === 0) {
+    bbox[0] = vec3.clone(vertex)
+    bbox[1] = vec3.clone(vertex)
+  } else {
+    vec3.min(bbox[0], bbox[0], vertex)
+    vec3.max(bbox[1], bbox[1], vertex)
+  }
+}
 
+/*
+ * Cache the bounding box of the geometry.
+ */
+const measureCached = (geometry, measureFn) => {
+  let boundingBox = cache.get(geometry)
+  if (boundingBox) return boundingBox
+  boundingBox = measureFn(geometry)
+  if (boundingBox.length === 0) {
+    // bounding box is undefined
+    boundingBox[0] = vec3.create()
+    boundingBox[1] = vec3.create()
+  }
   cache.set(geometry, boundingBox)
+  return boundingBox
+}
 
+/*
+ * Measure the min and max bounds of the given (path2) geometry.
+ * @return {Array[]} the min and max bounds for the geometry
+ */
+const measureBoundingBoxOfPath2 = (geometry) => {
+  const boundingBox = []
```

Comment:
So, why not allocate here, and remove all the unnecessary checks for length == 0 in measuredCached() and expand*() functions?

Reply:
So... I did it this way for a specific reason which is to maintain the current JSCAD behavior. Basically if there are no points, we historically return `[[0, 0, 0], [0, 0, 0]]` as the bounding box. But only if there are no points. In the old code, there was a specific check for `if (points.length === 0) {` in each of the measure functions.

But I can't allocate the bounding box vectors at the start, because otherwise the `expand` functions will improperly include `[0, 0, 0]`. The shape might not actually include the origin.

So I decided to treat empty list `[]` as "undefined bounding box" when there are no points yet. Why not `undefined`? Because I wanted to pass it to `expand` and allow expand to mutate the array.

I do wish the code was a little less "clever" in that sense. But I tried to make it as clear as I could, and I think overall it is much better than it was before. But I am open to suggestions.